js-miniapp-sample/src/pages/ads.js

```diff
@@ -0,0 +1,167 @@
+import React, { useReducer } from 'react';
+import MiniApp from 'js-miniapp-sdk';
+
+import {
+  Button,
+  CardActions,
+  CircularProgress,
+  Typography,
+  makeStyles,
+} from '@material-ui/core';
+import GreyCard from '../components/GreyCard';
+
+const useStyles = makeStyles((theme) => ({
+  content: {
+    height: '50%',
+    justifyContent: 'center',
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    fontSize: 18,
+    color: theme.color.primary,
+    fontWeight: 'bold',
+  },
+  actions: {
+    justifyContent: 'center',
+  },
+  error: {
+    marginTop: 10,
+  },
+}));
+
+type State = {
+  isLoading: ?boolean,
+  isError: ?boolean,
+};
+
+export const initialState = {
+  isLoading: false,
+  isError: false,
+};
+
+// $FlowFixMe
+export const dataFetchReducer = (state: State, action: Action) => {
+  switch (action.type) {
+    case 'LOADING':
+      return {
+        ...state,
+        isLoading: true,
+        isError: false,
+      };
+    case 'SHOW_SUCCESS':
+      return {
+        ...state,
+        isLoading: false,
+        isError: false,
+        reward: action.rewardItem,
+      };
+    case 'SHOW_FAILURE':
+      return {
+        ...initialState,
+        isLoading: false,
+        isError: true,
+      };
+    default:
+      throw Error('Unknown action type');
+  }
+};
+
+function Ads() {
+  const [interstitialState, interstitialDispatch] = useReducer(
+    dataFetchReducer,
+    initialState
+  );
+  const [rewardState, rewardDispatch] = useReducer(
+    dataFetchReducer,
+    initialState
+  );
+  const classes = useStyles();
+
+  const displayInterstitialAd = () => {
+    interstitialDispatch({ type: 'LOADING' });
+
+    const adUnitId = 'ca-app-pub-3940256099942544/1033173712'; // public test adId from Google.
+    MiniApp.loadInterstitialAd(adUnitId)
+      .then((loadSuccess) => {
+        console.log(loadSuccess);
+
+        MiniApp.showInterstitialAd(adUnitId)
+          .then((closedSuccess) => {
+            interstitialDispatch({ type: 'SHOW_SUCCESS' });
+            console.log(closedSuccess);
+          })
+          .catch((error) => {
+            interstitialDispatch({ type: 'SHOW_FAILURE' });
+            console.error(error);
+          });
+      })
+      .catch((error) => {
+        interstitialDispatch({ type: 'SHOW_FAILURE' });
+        console.error(error);
+      });
```

Comment:
Overall, I think you can simplify this block by returning the Promise inside of `then` and also chaining at the outer level:

```suggestion
    MiniApp.loadInterstitialAd(adUnitId)
      .then((loadSuccess) => {
        console.log(loadSuccess);

        return MiniApp.showInterstitialAd(adUnitId);
      })
      .then((closedSuccess) => {
        interstitialDispatch({ type: 'SHOW_SUCCESS' });
        console.log(closedSuccess);
      })
      .catch((error) => {
        interstitialDispatch({ type: 'SHOW_FAILURE' });
        console.error(error);
      });
```